# Copyright 2019 Google LLC
# 
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#  
#    https://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Dynamic Constraints
min_acceleration: -2.0
max_acceleration: 0.5
min_lateral_acceleration: -0.5
max_lateral_acceleration: 0.5
min_velocity: 0.0
max_velocity: 2.0
min_desired_velocity: 0.0
max_desired_velocity: 1.3

# Obstacle Constraints
robot_radius: 0.3
obstacle_check_front_offset: 0

# Planning parameters
planning_time_horizon: 6.0
time_bins: 21  # in interval {0, planning_time_horizon}
velocity_bins: 31  # in interval {min_velocity, max_velocity}
cycle_time: 0.1
reinitialization_distance: 1.0
max_occupancy_grid_value: 30
max_yaw_change_for_nonzero_velocity: 0.3
visited_grid_distance_resolution: 0.1
path_end_buffer_distance: 0.5

# Planning Cost Weights
travel_time_weight: 10
jerk_weight: 10.0
longitudinal_acceleration_weight: 100.0
lateral_acceleration_weight:  10.0
obstacle_weight: 1.0
desired_velocity_weight: 10.0

# Dead Zone Parameters
dead_zone_distance: 1.0
dead_zone_velocity: 0.1
dead_zone_deceleration: -0.5

# Timeouts
desired_path_timeout: 200
odometry_timeout: 0.2
perception_timeout: 0.5

# Obstacle Override Parameters
obstacle_override_timeout: 0.3
obstacle_override_max_velocity: 2.0
obstacle_override_desired_velocity: 0.7
